Sources/VSM/StateContainer+Debug.swift

```diff
@@ -49,9 +37,105 @@ private enum _StateContainerDebugLogger {
 extension StateContainer {
     func registerForDebugLogging() {
 #if DEBUG
-        if _StateContainerDebugLogger.isLoggingAllStateContainers {
-            _StateContainerDebugLogger.register(stateContainer: self)
+        if !StateContainerDebugLogger.defaultLoggingModes.isEmpty {
+            debugLogger.startLogging(for: self, options: StateContainerDebugLogger.defaultLoggingModes)
         }
 #endif
     }
 }
+
+/// Manages debug logging for a state container
+class StateContainerDebugLogger {
+    static var defaultLoggingModes: _StateContainerDebugOptions = []
+    private lazy var subscriptions: [_StateContainerDebugOptions: AnyCancellable] = [:]
+        
+    private struct Event<State> {
+        let name: String
+        let state: State
+        let description: String
+    }
+    
+    /// Registers debug logging once per mode per state container (to prevent duplicate logging from multiple calls)
+    func startLogging<State>(for container: StateContainer<State>, options: _StateContainerDebugOptions) {
```

Comment:
What happens if this gets called more than once, but with different options? Would you be reporting on multiple publishers debug info the same option(s)? 

What is the benefit to have a publisher _per_ option set, vs say, a publisher per option in the option set (ensuring you only have 1 per option)?

How do you stop logging?

Reply:
> What happens if this gets called more than once, but with different options? Would you be reporting on multiple publishers debug info the same option(s)?

There would be no duplicate log lines of the same options. The debug logger will print a debug line for each state change and for each call to `_debug(options:)` where `options` is unique.

> What is the benefit to have a publisher per option set, vs say, a publisher per option in the option set (ensuring you only have 1 per option)?

The latter would require code that dynamically combines multiple publishers, which would result in significantly more code. `willSet` and `didSet` options are also chronologically incompatible, which would cause other issues with logging and the timing of the state change.

> How do you stop logging?

There is no in-flight log-stop support. This feature was meant to function similar to SwiftUI's `_printChanges()` function. It is opt-in behavior that can only be stopped if the line is removed from code.